tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/InstrumentationDefinitionsGenerator.cs

```diff
@@ -729,8 +741,119 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                                  ? adoNetDefinitions
                                  : (adoNetDefinitions.IsDefaultOrEmpty ? definitions : definitions.AddRange(adoNetDefinitions));
 
-        string source = Sources.CreateCallTargetDefinitions(allDefinitions);
+        List<CallTargetDefinitionSource> orderedDefinitions;
+        string source = Sources.CreateCallTargetDefinitions(allDefinitions, out orderedDefinitions);
         context.AddSource("InstrumentationDefinitions.g.cs", SourceText.From(source, Encoding.UTF8));
+
+        if (dest.Tfm.Length > 0 && dest.Placeholders.Length > 0 && orderedDefinitions is not null)
+        {
+            GenerateNative(dest.Tfm[0], dest.Placeholders[0], orderedDefinitions, context);
+        }
+    }
+
+    private void GenerateNative(string tfm, string destFolder, IEnumerable<CallTargetDefinitionSource> orderedDefinitions, SourceProductionContext context)
+    {
+        var sb = new StringBuilder();
+        sb.Append(Datadog.Trace.SourceGenerators.Constants.FileHeaderCpp);
+        sb.AppendLine("""
+#pragma once
+#include "../../Datadog.Tracer.Native/generated_definitions.h"
+
+namespace trace
+{
+extern WCHAR* assemblyName;
+
+""");
+        GenerateSignatures(orderedDefinitions);
+        GenerateCallTargets(orderedDefinitions);
+        sb.AppendLine("""
+}
+""");
+
+        var filePath = Path.Combine(Path.Combine(destFolder, "Generated"), $"generated_callTargets_{GetTFMName()}.h".ToLower());
+        WriteAdditionalFile(filePath, sb.ToString());
+
+        string GetTFMName()
+        {
+            return tfm.Replace(".", "_");
+        }
```

Comment:
nit: Currently we're doing this replacement repeatedly (generating a bunch more strings). Might be nice to do it once and re-use

Reply:
I see your point

Reply:
Fixed in af7572f49550f1c3dbaad95fb44834e6b021784f